### ISSUE TITLE
Add rendering for man_made=adit

### ISSUE THREAD

**Issue by Adamant36** [gravitystorm/openstreetmap-carto#3704]
Rendering of man_made=adit was mentioned in https://github.com/gravitystorm/openstreetmap-carto/issues/3635#issuecomment-455189700 as a way to help the map appeal to more rural people. Plus, I just think it would be a cool thing to render. So I thought id open an issue for it. 

The tag currently has 9 ,136 uses, most of which are nodes. So the numbers are there. All it needs an icon, which conveniently has an example on the wiki for. 
![120px-mine1 svg](https://user-images.githubusercontent.com/30259065/53799913-69d2b300-3ef0-11e9-9384-4ea7d2d2803e.png)

Here's the usage chart also. It looks like it has had a pretty brisk increase in usage for a good number of years. 

![taghistory 3](https://user-images.githubusercontent.com/30259065/53800051-bc13d400-3ef0-11e9-9116-1e03d8d768e7.png)

**Comment by imagico** [comment]
Like with #3703 i would suggest to look into the practical use of the tag, in particular what spectrum of things the tag is used for and what other tagging concepts compete with that.

**Comment by matkoniecz** [comment]
@Adamant36 Can you do such review? https://taginfo.openstreetmap.org/tags/man_made=adit has link to overpass turbo query in top-right.

Let me know if you need some help with overpass or something else in this checks.

**Comment by jragusa** [comment]
Adits are usually not allowed for general public because of security reasons. When adit are closed, they are sealed by a gate.

Is it a good landmark ? I'm not convinced. First because they do not usually stand out as depicted in the picture of the wiki page. And secondly, because authority prefer to keep people away from this feature. BTW, I would be interested if such features are indicated on official topographic maps.

**Comment by matkoniecz** [comment]
> First because they do not usually stand out as depicted in the picture of the wiki page

Do you have a good image (I wanted to add it to Wiki as an additional one to not mislead about typical prominence). But from my trawling of https://commons.wikimedia.org/wiki/Category:Adits most adits with photos (I know, biased sample) are fairly prominent ones

**Comment by Adamant36** [comment]
> Like with #3703 i would suggest to look into the practical use of the tag

@imagico, I left a comment about it there. Maybe you could respond to it. In general, I think basing rendering on practical usage alone is a bad metric. Although, I will still look into it. I'm not an expert on adit's though and I'm sure a lot of these are tagged in fairly remote areas. So I'm not sure what I'll be able to take away from looking at practical usage. 

I don't think whoever decided to add an icon for caves was a spelunker and made sure the places tagged as caves weren't just small holes in the ground being wrongly tagged before rendering it. Although I could be wrong.

> most adits with photos (I know, biased sample) are fairly prominent ones

It seems like a lot of them have historical value also and can be important tourist spots or things to look at along hiking trails. I know they are where I live. 

> authority prefer to keep people away from this feature. 

I think they are neutral on the closed off ones. Since people can't get into them (probably open ones also in some cases). Some of them are part of tourist things also. As can be seen in the photos below. Maybe rendering them would the benefit of people adding access tags though, which could help mitigate some of that. It would be good to tell them apart from caves on the map also (maybe some adits are miss tagged as caves to render. Which rendering them could help fix). 

![adit](https://user-images.githubusercontent.com/30259065/53844052-60316580-3f59-11e9-8c33-4a037dee4b38.jpg)

![adit 2](https://user-images.githubusercontent.com/30259065/53844057-66bfdd00-3f59-11e9-967f-8e0f906f31ce.jpg)

![adit 4](https://user-images.githubusercontent.com/30259065/53844066-717a7200-3f59-11e9-8f64-5d36f7e816f1.jpg)

![adit 5](https://user-images.githubusercontent.com/30259065/53844075-7a6b4380-3f59-11e9-9e6e-ad4bd62b8c8e.jpg)

![adit 6](https://user-images.githubusercontent.com/30259065/53844085-822ae800-3f59-11e9-8435-af06a1e377c0.JPG)

**Comment by imagico** [comment]
You are free to look or not to look at use of tags and the tagging context.  My suggestions are meant to help you decide which tags make sense to render and what might be a suitable way to render them.

But it is equally fine if you just open the issue and let others more familiar with the matter look at things in more detail and decide on if and how to render this feature.

In the end looking at and familiarizing yourself with the nature of what you try to render (which is always the actual data in the OSM database and not some abstract concept of a tag) is an essential part of style development - and in many cases as important as the technical skills to implement your design idea.

**Comment by Adamant36** [comment]
> You are free to look or not to look at use of tags and the tagging context. My suggestions are meant to help you decide which tags make sense to render and what might be a suitable way to render them.

I know. I wasn't faulting you for suggesting it. I've been unsure what rendering metrics are best for a while now and anything helps. So I appreciate that you brought it up as an alternative. Generally though, I think its bad practice to not render something based only on instances of miss-tagging, not that I'm saying your advocating doing that here. Some metrics are more prone to bias then other's though and the more it can be eliminated the better in my opinion. 

Personally, in this instance I tend to lean more toward the tag being approved and therefore having wide community acceptance as reasons to render it, not so much what the actual usage is. Although I'm sure most of them are tagged properly. 

> But it is equally fine if you **just open the issue** and let others more familiar with the matter look at things in more detail and decide on if and how to render this feature.

I know I don't have any ultimate say in the decision since I'm not a maintainer, but can I not both open the issue and give reasons why I think the thing should be rendered? If its better to "just open the issue" and not add the extra details, I'm fine doing it that way instead. I don't want to do something if its not a help to anyone.

**Comment by wilmaed** [comment]
> All it needs an icon, which conveniently has an example on the wiki for.

![adit2](https://user-images.githubusercontent.com/31799538/53870938-7b54b180-3ffb-11e9-990b-efd6c657aa3c.png)
![adit2b](https://user-images.githubusercontent.com/31799538/53870946-80196580-3ffb-11e9-9d49-4a2d1343f97b.png)


https://github.com/wilmaed/foo/blob/master/adit_.svg

https://github.com/wilmaed/foo/blob/master/adit__.svg

**Comment by Adamant36** [comment]
@wilmaed, thanks. I can probably do a PR for it after the conversation about rendering it or not gets worked out.

**Comment by wilmaed** [comment]
Like a historical OSM-Map, I would only use the "Hammer and pick"-Icon, if there is a resource tag.

A bunker adit (without resource tag):
https://www.openstreetmap.org/node/2215161630

Source:
https://forum.openstreetmap.org/viewtopic.php?pid=322755#p322755

**Comment by Adamant36** [comment]
Hhmm, interesting discussion. I'm not sure what to make of it. I see that was in 2013 though. Haven't none mining tunnels been covered by the "tunnel" tag since then though? I doubt miss use of adit for those other things, like military bunkers if I read it correctly, is really that high anymore.

**Comment by wilmaed** [comment]
Not every mine/adit is for mining.
The german community will tag this Luftschutzstollen (Luftschutzstollen Gaishalde) as an adit:
![bibi_luftschutzstollen_gaishalde](https://user-images.githubusercontent.com/31799538/53881448-05107900-4014-11e9-8102-2b9f515a7fa4.jpg)
Source:
https://de.wikipedia.org/wiki/Luftschutzstollen
https://www.openstreetmap.org/node/1894748295
https://de.wikipedia.org/wiki/Luftschutzstollen#/media/File:Bibi_Luftschutzstollen_Gaishalde.jpg

An adit with 2 entrances is still an adit and not a tunnel (even if you use one entrance as an exit).

**Comment by matkoniecz** [comment]
@wilmaed Can you document this on OSM Wiki? If there is doubt is it correct tagging or a mistake I recommend discussion on tagging mailing list or equivalent.

**Comment by Adamant36** [comment]
> An adit with 2 entrances is still an adit and not a tunnel (even if you use one entrance as an exit).

So then its miss-tagging to tag these none mining things as adits? It seems like they are adits, but then it also seems like they aren't. I'm confused :confused:. You'll have to link to the mailing list discussion if you discuss it there.

**Comment by wilmaed** [comment]
> So then its miss-tagging to tag these none mining things as adits? 

Only if you define an adit "only for mining resources".
This is too strict for these historic objects. 

> You'll have to link to the mailing list discussion if you discuss it there.

I will not jump into a snake pit or touch a nest of vipers :)

Guess: adit for resources or an adit to protect people:

![50308309_396828481050074_1679100603580070976_n](https://user-images.githubusercontent.com/31799538/53888625-4b220880-4025-11e9-923d-7927484e793a.jpg)

Luftschutzstollen (bunker)

If it looks like a duck, swims like a duck, and quacks like a duck, then it probably is a duck/adit.

**Comment by wilmaed** [comment]
natural=cave_entrance 
is misused for adits:

https://www.openstreetmap.org/node/1449303526
https://www.openstreetmap.org/node/1449303524
...

These adits/Luftschutzstollen are man_made (not natural):
http://www.geheimprojekte.at/ls_ternitz-blindendorf.html

**Comment by wilmaed** [comment]
@Adamant36
A proposal for an extended definition exists in OSM Talk:
https://wiki.openstreetmap.org/wiki/Talk:Tag:man_made%3Dadit 

> There are adits for non-mining purposes, especially **_old air raid shelters_** ...

> I propose to change "An adit is a (nearly) horizontal entrance to the underground, by which resources can be extracted or a mine can be entered, ventilated or drained of water. 
>The visible portal of the adit is mapped as a node." to "An adit is a artificial, (nearly) horizontal entrance to the underground. Most adits are built as a means to extract resources, drain a mine of water or provide access to other parts of the mine. The visible portal of the adit is mapped as a node." (edits in italics). 
>See, for reference, the article of the german wikipedia, which describes an adit as a means to access a Grubenbau, which is defined as "any subsurface cavity created by miners", **_regardless of purpose_** of said Grubenbau. --Gormo (talk) 11:49, 30 April 2018

**Comment by Adamant36** [comment]
> I will not jump into a snake pit or touch a nest of vipers :)

Can't say I blame you there. 

So are you against the hammer and pick for none mining adits? It seems like its a universally recognized symbol for them. Even for none mining ones (most of which use to be mines at one point from my research). Plus, it probably took those tools to dig a lot of those places originally. 

> A proposal for an extended definition exists in OSM Talk:

So where are we at with this then? I'm still ewing and awing over the pictures.

**Comment by jeisenbe** [comment]
> Adits are usually not allowed for general public because of security reasons. When adit are closed, they are sealed by a gate.

Yes, small mines are shown on USGS topo maps, with this symbol.

> Not every mine/adit is for mining. The german community will tag a Luftschutzstollen (Luftschutzstollen Gaishalde) as an adit

In British English, the word adit is only used for a horizontal
entrance to a mine: https://en.oxforddictionaries.com/definition/adit

However, I understand that the usage of similar words in other
languages can affect the use of tags.

In this case, since the German adits are also made by mining, even
though not all are for mineral extraction, I believe the same symbol
is appropriate. The crossed mattock and hammer represents the tools
used by miners to create the shaft or adit, not the produce of the
mine.

We should also consider if the same rendering should be used for
man_made=mineshaft (8849 uses) and historic=mineshaft (7 745 uses)
https://taginfo.openstreetmap.org/tags/man_made=mineshaft
https://taginfo.openstreetmap.org/tags/historic=mine_shaft

**Comment by Adamant36** [comment]
> We should also consider if the same rendering should be used for
man_made=mineshaft (8849 uses) and historic=mineshaft (7 745 uses)

I was thinking that myself. The symbol would probably work for all three. I had actually planned to open issues for them both after this one either gets rejected or rendered, but I didn't want to bring them up here necessarily so the conversation can stay as on topic as possible and devolve to a discussion about other tags or whatever. Not that I'm saying I have anything against you bringing them up here though.

**Comment by jeisenbe** [comment]
@Adamant36, were you still considering submitting a PR for this? While there is some disagreement about how many additional svg icon symbols can be added to the map, the great majority of adits and mine shafts are in rural areas where crowding is not a big problem, and I believe this symbol will be quite intuitive in most places. So this would be an example of a symbol that I would support adding (rather than, say, yet another type of shop).

**Comment by Adamant36** [comment]
Yeah, id love to see them rendered. So id 100% do it, but I don't feel like putting in the time or effort if @imagico is just going to shot it down. So I'm waiting until the icon issue is dealt with before putting the work into it. I'm totally willing to do before then though if @imagico gives his ok on it. I don't see any reason why he can't make a judgement about its merits intially, before its a fully tested PR.

**Comment by imagico** [comment]
Independent of the question of the long term strategy on point symbol rendering in this style (see #3635) i can mainly point to what i wrote in https://github.com/gravitystorm/openstreetmap-carto/issues/3703#issuecomment-469659265.  An analysis of actual use of the tag in OSM as well as semantically related tags (like the mentioned `man_made=mineshaft` and `historic=mine_shaft` but also, `historic=mine` - 6500 uses, `man_made=mine` - 6000 uses, mostly in the US and `landuse=quarry` - 36k uses on nodes) would be an essential prerequisite for deciding on rendering it in this style as well as suitability of possible symbols for this.

**Comment by jeisenbe** [comment]
@Adamant36, I agree that it would be nice to check how the feature is
being used, since there were also concerns about a number of tunnels,
bunkers and  underground storage features which are called adits in
German-speaking areas. If you use JOSM to download all of the
man_made=adit (and perhaps the mineshaft / mine_shaft features) you
could check where tag is used in a few different countries.

It would be interested to check some of the nodes tags landuse=quarry
too - are many of these mistagged adits or mineshafts? Probably they
are just quarry featues that haven't yet been mapped as areas, but it
would be interesting to check.

I know this takes more time, but it can also be fun

**Comment by Adamant36** [comment]
> I agree that it would be nice to check how the feature is
being used

Yeah, I seem to remember that being an issue before. I'll check it out when I get some time. I actually got in a discussion with someone on the wiki because they changed the historic=mine page to say they could be mapped on areas. I had assumed a historic mine as an area would just be a quarry. It's really clear from the wiki if landuse=quarry is suppose to be used on only above ground mining operations or if it fits for older filled in underground mines though. So, its something I wanted to do more research on anyway.

**Comment by Adamant36** [comment]
So I checked things out. Like 3001 out of  9,504 of them have disused=yes and it seems like most of those are in Germany. Along with a good number of the 827 tagged as historic. Most of the adit's in Germany looked like normal ones. Including those. Except there a few that were tagged on tunnels, but that was only on ways. There were also a couple miss-tagged on quarries, but only as areas, on one island somewhere, and I left a note for someone to fix them. I didn't see any storage facilities tagged as adits. There was a couple of buildings, but they seemed like legitimate mining buildings that probably had the adit inside of them. I think if we just render nodes and use the upside down hammer and pick for disused ones we should be fine. 

As far as quarries go. I didn't check as many out as I did the adits, but it seems like with nodes the vast majority of them are normal quarries that just haven't been mapped as areas yet. A few were historical mines in heavily wooded areas, but it's not enough to be an issue. None were miss-tagged adits or mineshafts from what I saw.

**Comment by jeisenbe** [comment]
Thank you. Have you had a chance to look at the usage of the mineshaft tags
(historic= and man_made)? I imagine that these would use the same icon.

**Comment by jragusa** [comment]
Opacity could be an alternative to display differently disused adits if people are not aware about the upside down icon.

**Comment by Adamant36** [comment]
True. People might confuse opacity with private access though. From what the wiki said the upside thing is semi well known. Although I don't think that means we go with it though if there's better alternative, because really who knows.

**Comment by jeisenbe** [comment]
Since many audits are disused and some were never used for mineral
extraction, I believe it is ok to use one symbol for all adits: the feature
is the mined horizontal tunnel, rather than the active presence of certain
mineral extraction activity.
Similarly, I would be support rendering disused mine shafts.
We also render disused quarries/open surface mines the same as active
quarries/mines, so there is precedent.

**Comment by Adamant36** [comment]
>  Have you had a chance to look at the usage of the mineshaft tags
(historic= and man_made)? I imagine that these would use the same icon.

No I haven't. Well, I did look at a few instances of man_made=mineshaft on areas and it seemed like they were overwhelmingly wrongly tagged on the area of the whole quarry. Which is unfortunate. I didn't do enough research to call it conclusive though. Maybe we could save both for another issue when I have time to look into it more extensively and just deal with adits for now.

**Comment by Adamant36** [comment]
@jeisenbe, what do you think the next step on this is?

**Comment by jeisenbe** [comment]
Try test renderings of man_made=adit and perhaps also historic= and man_made= mineshafts would be helpful.

As suggested above it would also be good to check out how the tag is being used in different places, not just in California or North America but also in some other countries.

**Comment by Adamant36** [comment]
> As suggested above it would also be good to check out how the tag is being used in different places, not just in California or North America but also in some other countries.

When I checked some before it was all in Europe from what I remember.

**Comment by ppete2** [comment]
Great idea to render man_made=adit. As indicated on the photos of @Adamant36 of 6 Mar 2019, these are sometimes interesing, historical landmark to visit, sometimes also with information plates nearby.
So I would welcome rendering these landmarks. Symbols could be those 2 of the Wiki site: https://wiki.openstreetmap.org/wiki/Tag:man_made%3Dadit

**Comment by pagaille** [comment]
I'm also in favour of displaying those landmarks (adits, cave entrances, ...) on the map. They are parts of our environment exactly like, say, power poles. I believe that some of them were displayed until recently but I may be wrong. 

Yes, public mapping organisations do map them.

Regarding the icons that could be displayed JOSM already has them if you're looking for some inspiration.

**Comment by daholzfeind** [comment]
I agree with @pagaille. I was checking out the notes in some place where a lot of adits are located, none of them is rendered, resulting in a lot of notes, c.f. https://www.openstreetmap.org/note/3504018

Rendering could be similar to the OpenTopoMap: https://opentopomap.org/#map=17/46.63063/13.65431

**Comment by HolgerJeromin** [comment]
> I agree with @pagaille. I was checking out the notes in some place where a lot of adits are located, none of them is rendered, resulting in a lot of notes, c.f. https://www.openstreetmap.org/note/3504018

This very note is from streetcomplete which has an own rendering style (aka not osm-carto) with much fewer POIs to show.

**Comment by daholzfeind** [comment]
@HolgerJeromin I did check it on OSMAnd now. Indeed with osm-carto the adits in the note location are displayed, even though the icon is a bit confusing on the first glance. 

Now the remaining question is, if osm.org uses osm-carto, why the adits are not rendered there.

**Comment by Erdwinger** [comment]
I´d also prefer that there was a rendering for reasons already mentioned above.
I suggest using ![adit in general](https://wiki.openstreetmap.org/w/images/1/15/Stollen2.png) for adit in general, ![adit + mine](https://wiki.openstreetmap.org/w/images/c/c2/Stollen.png) for adits to a mine, and ![adit + disused mine](https://wiki.openstreetmap.org/w/images/f/f3/Stollen-d.png) for adits to a disused mine. 
Using solely ![mine](https://user-images.githubusercontent.com/30259065/53799913-69d2b300-3ef0-11e9-9384-4ea7d2d2803e.png) might cause confusion, as it is associated with mine in a stricter sense, but there are mines without adits and also adits without mines. 

Source: [https://wiki.openstreetmap.org/wiki/Historical_Objects/Map_Properties](url)

**Comment by HolgerJeromin** [comment]
@Erdwinger our icons are 14 px
Does your icon works in this small size?

https://github.com/gravitystorm/openstreetmap-carto/blob/master/CONTRIBUTING.md#map-icon-guidelines

**Comment by Erdwinger** [comment]
Thanks @HolgerJeromin for the note and the link! (I´m new to OSM...)
I will try creating SVGs and also post them for checking. But I guess it should work as there already are icons similarly detailed.

**Comment by ppete2** [comment]
@Erdwinger : Any news on creating small-sized icons for adits?

**Comment by ldreier** [comment]
I created some svg symbols to progress discussion.
I looked at the proposals of @Erdwinger and also to the German official map standard
The commit https://github.com/ldreier/openstreetmap-carto/commit/cb2812eeb4239b51497d8f149a9b6a8f70710abb contains the 14x14 SVG files.

Overview:

| tag=value (proposed) | common proposal | German Topo Maps | Existing
:-----------------:|:--------------------------:|:-----------------------------:|:-----------:
natural=cave_entrance | | ![Image](https://github.com/user-attachments/assets/281a0547-9e99-4ff8-b48a-83c7e4825f74)| ![Image](https://github.com/user-attachments/assets/0aa00c7b-8e66-4fc9-a338-271086742a0d)
man_made=adit|![Image](https://github.com/user-attachments/assets/595ca6b0-bc96-40d1-b0e0-53d585caf6c5)| ![Image](https://github.com/user-attachments/assets/6cd88f16-b2ae-4fce-9bae-3dc31481f117)|
man_made=adit, operational |![Image](https://github.com/user-attachments/assets/0f0e53c2-13c5-44de-8533-ca5f8b7fdead)||
man_made=adit, disused |![Image](https://github.com/user-attachments/assets/5f7e889c-66da-4b50-975b-816f9d0ac6e3) ||
man_made=mineshaft | ![Image](https://github.com/user-attachments/assets/15498388-daa2-4a56-97ea-ad33633e742c) | ![Image](https://github.com/user-attachments/assets/3a454887-1876-47a8-b0d3-6fdc29c047f4) | 
man_made=mineshaft, disused | ![Image](https://github.com/user-attachments/assets/f72b4770-fd0a-4cd2-8cab-0dfd50641c60) |same as above |
industrial=mine, operational||![Image](https://github.com/user-attachments/assets/15498388-daa2-4a56-97ea-ad33633e742c)|
industrial=mine, disused|| ![Image](https://github.com/user-attachments/assets/f72b4770-fd0a-4cd2-8cab-0dfd50641c60) |

We need to choose a subset of the proposed symbols for the map.

**Comment by imagico** [comment]
This is very valuable, taking a holistic look at the cartographic context is always a good idea.

Note that we currently already use the *hammer and pick* symbol in the pattern for `landuse=quarry`.  This would need to be taken into account when developing an overall concept for rendering mining features.  We currently render `landuse=quarry` nodes (>30k uses) with a label only, the symbol is only used for the polygon fill pattern.  The pattern design could of course change, in particular it would be possible to visualize `resource=*` and/or the `disused` status.

Another important question to consider is if a third symbol variant is needed apart from *operational* and *disused*.  You have this for `man_made=adit` but not for `man_made=mineshaft`. On `man_made=adit` and `man_made=mineshaft` `disused=*`/`abandoned=*` is tagged on ~30 percent of features.

And keep in mind that there are additional tags in use:

* `historic=mine_shaft` (11k uses)
* `historic=mine` (13k uses)
* `man_made=mine` (3.4k uses, mostly from import in the US)

A more concrete suggestion: It might be worth trying to visualize `man_made=mineshaft` similar to the adit symbols with a compound symbol showing both physical form and function.

**Comment by dch0ph** [comment]
> We currently render `landuse=quarry` nodes (>30k uses) with a label only, the symbol is only used for the polygon fill pattern. The pattern design could of course change, in particular it would be possible to visualize `resource=*` and/or the `disused` status.

Yes, it would be excellent to render disused status of quarries. Abandoned quarries can be nature reserves / publicly accessible, which makes them very different features to active quarries. The approach I've taken in my fork is to use the "scrub green" for the fill colour e.g, this long-term abandoned quarry

(Carto)

<img width="204" height="159" alt="Image" src="https://github.com/user-attachments/assets/4396876f-1570-4e1e-8ce3-0ae033184ec1" />

(Fork)

<img width="159" height="130" alt="Image" src="https://github.com/user-attachments/assets/deba6c4f-cc29-43bc-a6c8-ef7785837dad" />

Stats:
`disused:landuse=quarry`: 1.6k
`abandoned:landuse=quarry`: 2k

`landuse=quarry`: 243k, of which
`disused=yes`: 7.2k
`abandoned=yes`: 3.6k


> Another important question to consider is if a third symbol variant is needed apart from _operational_ and _disused_. You have this for `man_made=adit` but not for `man_made=mineshaft`.

Yes, I'm puzzled why there are three symbols for `man_made=adit`, given that there isn't a tag for "operational". 
 
I'm a bit wary about the "upwards" vs. "downwards" tools distinction, which is not immediately intuitive (bearing in mind that we have no legend). It's even harder to distinguish to small versions combined with the adit symbol. One option might be to make the downwards tools a lighter shade of grey. We don't currently have symbols that have multiple fill colours, but I don't think there is a technical problem with this?

**Comment by dch0ph** [comment]
> Just an interim comment...

Sure. Rendering disused quarries would be a separate issue and PR. But it is useful to consider design issues in the round. Otherwise you end up with a mish-mash of different design choices.

**Comment by ldreier** [comment]
Here is now a more concrete proposal for rendering mine shafts.
For reference, the code changes are in 6ef0268cd5ef72efcc64ac3bdb423ea9affc8c55 this is not a "clean" pull request since the related branch also include my changes to get the docker containers up and running and also may need some more compact formatting for SQL.

This proposes rules for presenting "mallet and iron" symbols both in "normal" and "top-down" version. Latter is for "disused" mine shafts.
Areas/polygons which are mainly marked as man_made=mineshaft are now also rendered with the same grey as "quarry" (but without the repeating pattern)

For the "disused" symbol, tags like "abandoned=yes" or "disused=yes" or "historic=mine_shaft" are checked.

Here are some example results:

<img width="199" height="133" alt="Image" src="https://github.com/user-attachments/assets/bfe9d8c3-c890-4a1f-9501-8a5cfeb60062" />

<img width="290" height="274" alt="Image" src="https://github.com/user-attachments/assets/2eef7ef8-6239-4ea9-8fda-97ae9692b881" />

<img width="303" height="251" alt="Image" src="https://github.com/user-attachments/assets/a6aebb75-abce-4f72-8430-a33dc48235dc" />

<img width="1120" height="645" alt="Image" src="https://github.com/user-attachments/assets/46b462c6-8cab-4ad8-9ec7-b24de385bfaf" />

<img width="1025" height="566" alt="Image" src="https://github.com/user-attachments/assets/f7d3f64d-9714-41a5-8262-558649b94d41" />

<img width="578" height="682" alt="Image" src="https://github.com/user-attachments/assets/a1a9e4d1-74d8-436e-842b-70c285d6750c" />

<img width="708" height="518" alt="Image" src="https://github.com/user-attachments/assets/9d98ac09-9614-4e46-a45b-6fd351052fb6" />

<img width="1165" height="351" alt="Image" src="https://github.com/user-attachments/assets/fa761e87-371d-4163-b22e-e7aac2098ba7" />

In a next step I would look on mines (man_made=mine, historic=mine etc.) to be rendered with the same symbols.

**Comment by imagico** [comment]
That goes into the direction i had in mind.  But of course the symbol size is so small it will often not be recognizable.

And the question for OSM-Carto of course remains if the vertically mirrored symbol is a globally intuitive depiction of disused status.  Other design approaches would be possible as well - the adit symbol could be combined with the depiction of a minecart to indicate operational status and for mineshaft a headframe could be depicted.  I am not saying this is necessarily better, just that there are options that could be explored.

And ultimately it needs to go towards an actual style change.

**Comment by dch0ph** [comment]
> And the question for OSM-Carto of course remains if the vertically mirrored symbol is a globally intuitive depiction of disused status.

As noted above, I don't think it is. So it would need to be reinforced in some way, e.g. a different / no fill colour for disused quarry.

I do think it is creating extra design work by have 3 states (used, disused, unspecified) for some keys, and it would be simpler, e.g. to only show the hammer and pick when resource is specified (which seems a sensible proxy for an active mine/adit etc.)

**Comment by ldreier** [comment]
Here is a design for a headframe...

![Image](https://github.com/user-attachments/assets/c53bd345-bf1a-4c05-b4dd-c3dd11252eda)

As a starting point I took a more detailed sketch:

<img width="124" height="228" alt="Image" src="https://github.com/user-attachments/assets/f28c929f-adf8-4ce5-a886-fb6fd2d26f85" />

... but due to the limited size, I removed most of the details...

Also worth to note that there are multiple headframe configurations in reality, which may deviate from this steel box-profile tower with two carriages and  Koepe-type winder.
For further reading: https://www.foerdergerueste.de/geruestarten.htm

Example map:

<img width="240" height="151" alt="Image" src="https://github.com/user-attachments/assets/553316f9-20bb-4458-92ae-b99c152be053" />

Within the small symbol area I see no good way of depicting a operational vs. disused mineshaft. With more space I could think of:
- Showing the cables (or not)
- Showing the mineshaft ground-level opening as open or covered
- (and the usual proposal of adding the the "mallet and iron" in normal and rotated version)

**Comment by dch0ph** [comment]
> [@dch0ph](https://github.com/dch0ph) - I don't think a simple _hole in the ground_ illustration would be recognizable.

![Image](https://github.com/user-attachments/assets/ec9d3b04-174f-4b3b-895f-e7c54079d3cd)

This was the kind of thing I had in mind - a shaft with some "ticks" suggesting solid ground. This could be combined with a pit wheel for an active mineshaft.

<img width="14" height="14" alt="Image" src="https://github.com/user-attachments/assets/3c4f2559-7639-44f6-aeca-4716f713b5bd" />

I like the "end on carts". The "ticks" in this version make it look less like a simple arch (which could be unclear when there is no mine cart), It might look better for the "ticks" to point outwards rather than inwards to suggest going into rock?

I'm afraid the headframe symbols above aren't working for me. They look a bit like a lollipop balanced against a glass. I think a larger pit wheel might work better. The wheel is the most distinctive feature of a headframe.

But I do like the way things are going!

**Comment by imagico** [comment]
Nice work in exploring the different options.  As indicated the _hole in the ground_ idea is difficult to make recognizable without additional visual hints about the meaning.  

I had a go at this myself and here are a number of design concepts i came up with.  The headframe versions for the mineshaft, however, do not fit into the 14x14 pixel frame (though they are not very heavy in terms of optical weight).

| | plain | disused | abandoned
| --- | --- | --- | ---
| adit | ![adit](https://raw.githubusercontent.com/imagico/osm-carto-alternative-colors/refs/heads/master/symbols/sources/man_made/adit.svg) | ![adit disused](https://raw.githubusercontent.com/imagico/osm-carto-alternative-colors/refs/heads/master/symbols/sources/man_made/adit_disused.svg) | ![adit abandoned](https://raw.githubusercontent.com/imagico/osm-carto-alternative-colors/refs/heads/master/symbols/sources/man_made/adit_abandoned.svg)
| mineshaft | ![mineshaft](https://raw.githubusercontent.com/imagico/osm-carto-alternative-colors/refs/heads/master/symbols/sources/man_made/mineshaft.svg) | ![mineshaft disused](https://raw.githubusercontent.com/imagico/osm-carto-alternative-colors/refs/heads/master/symbols/sources/man_made/mineshaft_disused.svg) | ![mineshaft abandoned](https://raw.githubusercontent.com/imagico/osm-carto-alternative-colors/refs/heads/master/symbols/sources/man_made/mineshaft_abandoned.svg)
| mineshaft headframe 1 | ![mineshaft headframe 1](https://imagico.de/files/mineshaft_headframe1.svg) | ![mineshaft headframe 1 disused](https://imagico.de/files/mineshaft_headframe1_disused.svg) | ![mineshaft headframe 1 abandoned](https://imagico.de/files/mineshaft_headframe1_abandoned.svg)
| mineshaft headframe 2 | ![mineshaft headframe 2](https://imagico.de/files/mineshaft_headframe2.svg) | ![mineshaft headframe 2 disused](https://imagico.de/files/mineshaft_headframe2_disused.svg) | ![mineshaft headframe 2 abandoned](https://imagico.de/files/mineshaft_headframe2_abandoned.svg)

**Comment by ldreier** [comment]
@imagico 
quick comment on "mineshaft headframe 2" ... this does not feel like a "typical headframe". (see also link above https://www.foerdergerueste.de/geruestarten.htm). In the linked page, this shape matches a medieval hand-operated winch.
Typically the strongest pillar is dividing the angle of the cable by 2 ("bisector") to mainly have forces in the direction of this pillar when the cable is put under load. The other pillar is typically vertically mounted. Some headframes which have symmetric winders on both sides have the "bisector" pillars on both sides (like https://de.wikipedia.org/wiki/Datei:Zeche_Zollverein_-_Essen_-_letzter_Arbeitstag_23_12_1986.jpg)
